src/registry/types.rs

```diff
@@ -4,38 +4,43 @@ use std::sync::Arc;
 
 use rustc_serialize::json::{Json, ToJson};
 
-/// How much access things have
-#[derive(Debug, PartialEq, Eq, Clone, Copy)]
-pub enum RegistryAccess {
-    Public,
-    Lua,
-    Private
+bitflags! {
+    /// Access permissions for items in the registry
+    pub flags AccessFlags: u8 {
+        const LUA_PRIVATE = 0b00000000,
+        const LUA_ACCESS  = 0b00000001,
+        const LUA_WRITE  = 0b00000010,
```

Comment:
They're flags (the convention is `1 << 0,  1 << 1, 1 << 2`) but I like how the binary representation shows where the ones are. It would look better aligned though.